packages/@uppy/dashboard/src/Dashboard.tsx

```diff
@@ -250,12 +252,18 @@ export default class Dashboard<M extends Meta, B extends Body> extends UIPlugin<
 
     this.defaultLocale = locale
 
-    // Dynamic default options:
-    this.opts.doneButtonHandler ??= () => {
-      this.uppy.clearUploadedFiles()
-      this.requestCloseModal()
+    // support for the legacy `autoOpenFileEditor` option,
+    // TODO: we can remove this code when we update the Uppy major version
+    let autoOpen: false | 'imageEditor' | 'metaEditor'
+    if (!opts) {
+      autoOpen = false
+    } else if (opts.autoOpen === undefined) {
+      autoOpen = opts.autoOpenFileEditor ? 'imageEditor' : false
+    } else {
+      autoOpen = opts.autoOpen
     }
-    this.opts.onRequestCloseModal ??= () => this.closeModal()
+    // merge default options with the ones set by user
+    this.opts = { ...defaultOptions, ...opts, autoOpen }
```

Comment:
Merging is now done in the `super` call.

```suggestion
    this.opts.autoOpen = autoOpen
```

Reply:
Yep, was reconciliating the merge in 2 softwares, I think one overrode the other